knora/dsplib/utils/xml_upload.py

```diff
@@ -574,6 +585,44 @@ def validate_xml_against_schema(input_file: str, schema_file: str) -> bool:
     return is_valid
 
 
+def convert_ark_v0_to_resource_iri(ark: str) -> str:
+    """
+    Converts an ARK URL from salsah.org (ARK version 0) of the form ark:/72163/080c-779b9990a0c3f-6e to a DSP resource
+    IRI of the form http://rdfh.ch/080C/Ef9heHjPWDS7dMR_gGax2Q
+
+    This method is needed for the migration of projects from salsah.org to DSP. Resources need to be created with an
+    existing ARK, so the IRI needs to be extracted from that ARK in order for the ARK URL to be still valid after the
+    migration.
+
+    Args:
+        ark : an ARK version 0 of the form ark:/72163/080c-779b9990a0c3f-6e, '72163' being the Name Assigning Authority
+        number, '080c' being the project shortcode, '779b9990a0c3f' being an ID derived from the object's Salsah ID and
+        '6e' being check digits
+
+    Returns:
+        Resource IRI (str) of the form http://rdfh.ch/080C/Ef9heHjPWDS7dMR_gGax2Q
+    """
+    # create the DaSCH namespace to create version 5 UUIDs
+    generic_namespace_url = uuid.NAMESPACE_URL
+    dasch_uuid_ns = uuid.uuid5(generic_namespace_url, "https://dasch.swiss")  # cace8b00-717e-50d5-bcb9-486f39d733a2
+
+    # get the salsah resource ID from the ARK and convert it to a UUID version 5 (base64 encoded)
+    if ark.count("-") != 2:
+        raise BaseError(f"while converting ARK '{ark}'. The ARK seems to be invalid")
+    project_id, resource_id, check_digits = ark.split("-")
+    project_id = project_id.split("/")[-1].upper()
```

Comment:
some cosmetic remarks:
- I like to use `_` as a variable for values that are discarded (like the check digits)
- if interested in the last element(s) of a split, it's usually more elegant to use `rsplit()`
- for both `split()` and `rsplit()` it can be useful to limit the maximum number of splits you want, so that you can be sure of the number of results you get, and then unpack into variables more easily (so you can do things like `_, second_last, last = long_string.rsplit("-", 2)`)